src/sinks/axiom.rs

```diff
@@ -150,17 +163,35 @@ impl SinkConfig for AxiomConfig {
 
 impl AxiomConfig {
     fn build_endpoint(&self) -> String {
-        let url = if let Some(url) = self.url.as_ref() {
-            url.clone()
-        } else {
-            CLOUD_URL.to_string()
-        };
+        // Priority: url > region > default cloud endpoint
+
+        // If url is set, check if it has a path
+        if let Some(url) = &self.url {
+            let url = url.trim_end_matches('/');
+
+            // Parse URL to check if path is provided
+            // If path is empty or just "/", append the legacy format for backwards compatibility
+            // Otherwise, use the URL as-is
+            if let Ok(parsed) = url::Url::parse(url) {
+                let path = parsed.path();
+                if path.is_empty() || path == "/" {
+                    // Backwards compatibility: append legacy path format
+                    return format!("{}/v1/datasets/{}/ingest", url, self.dataset);
```

Comment:
nit: 

```suggestion
                    return format!("{url}/v1/datasets/{}/ingest", self.dataset);
```

Same for all format! calls.